packages/zosfiles/src/methods/search/Search.ts

```diff
@@ -100,14 +102,33 @@ export class Search {
                 const response = await List.allMembers(session, pds, searchOptions.listOptions);
                 if (response.apiResponse.items.length > 0) {
                     for (const item of response.apiResponse.items) {
-                        if (item.member != undefined) { searchItems.push({dsn: pds, member: item.member}); }
+                        if (item.member != undefined) { searchItemsQueue.push({dsn: pds, member: item.member}); }
                     }
                 }
             } catch (err) {
                 failedDatasets.push(pds);
             }
         }
 
+        // Call the callback if it exists.
+        let resp: boolean = true;
+        if (searchOptions.continueSearch) {
+            // Call with a clone of the search items queue. Extenders should not be modifying the original.
+            resp = await searchOptions.continueSearch(cloneDeep(searchItemsQueue));
```

Comment:
Since the callback is someone else's code (which might crash), should we put a try-catch safety net around the callback? Certainly a stack trace will identify if the failure occurred in the callback, but the failure could be perceived as a failure in our app/API.

I know we have numerous other callbacks, and maybe they do not check for crashes, but maybe we should start? This is not a hard request for change. If you consider the idea and make a different choice, that is ok.

Reply:
How would we want to handle a crash? A graceful search termination, or just re-throwing it?

Reply:
Spoke with many members of the team. We decided to let the code throw any failed callback, and expect anyone using the search APIs to handle their own errors. If a callback does throw, the error should point directly to the callback function.